x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/authc/saml/SamlAuthenticator.java

```diff
@@ -395,4 +405,18 @@ private void checkLifetimeRestrictions(Conditions conditions) {
     private void checkLifetimeRestrictions(SubjectConfirmationData subjectConfirmationData) {
         validateNotOnOrAfter(subjectConfirmationData.getNotOnOrAfter());
     }
+
+    private void warnOnReservedTermAttributeNames(Assertion assertion, List<Attribute> attributes) {
```

Comment:
I kept the name pretty generic; good with going with something more nameId specific instead

Reply:
I am OK with more general naming. But the wording we used for these names are ["Special attribute names"](https://www.elastic.co/guide/en/elasticsearch/reference/current/saml-guide-stack.html#saml-attribute-mapping-nameid) instead of "reserved term". I am also ok with calling it "synthetic attribute name" since that is the word used in the code. We will need adjust the logging message accordingly as well.

Reply:
Makes sense, going with special attribute name!